src/components/icon/icon.svelte

```diff
@@ -4,13 +4,23 @@
 </script>
 
 <script lang="ts">
-  export let name: string = undefined
+  import type { IconNames, StringWithAutocomplete } from '../types'
+
+  export let name: StringWithAutocomplete<IconNames> = undefined
   export let forceColor: boolean = false
+  export let size: string = 'unset'
+  export let color: string | undefined = 'unset'
```

Comment:
Can you support this with an example of where this is more better than defining size and color in css? Every other component / html element is styled via css (or inline style if you must!), with all the features css provides (cascade, media queries, hover and focus states). If this is a developer-efficiency thing, then I'm just struggling to think of scenarios where there isn't at least some CSS in an app, unless it's a prototype. Or solved with tailwind, because that would provide similar functionality to every element, e.g. `<p class="color-red hover:color-darkred">`. With regular css, it doesn't even need to be provided to just this element but can be provided by a parent, e.g.

```css
.toolbar {
  --leo-icon-size: 24px;
  color: blue;
}
```
or
```css
.toolbar {
  --leo-icon-size: 32px;
  @container (width < 321px) {
    --leo-icon-size: 24px;
  }
}
```
or
```css
.icon {
  color: pink;
  &:hover {
    color: blue;
  }
}
```
or
```css
.indicator {
   &.isError .icon {
      color: red;
   }
   &.isWarning .icon {
      color: yellow;
   }
}
```

Reply:
The CSS approach would also mean that we don't need a special `darkModeColor` prop since the consumer can change the color via its own media queries however it pleases.

Reply:
This is how I am seeing Icon instance colors and sizes getting set from within React.

![Screenshot 2023-03-23 at 2 05 17 PM](https://user-images.githubusercontent.com/30185185/227304714-43bf6da2-fa43-4f5a-b96c-edc05d0c0b27.png)

Reply:
Seems like that should work. Is the issue that it's very verbose and that you have to make a new element for each time we want to change the color or size of something? If so, that's a problem I too have with styled-components in general - same is true every time we want a different font-size or color for a piece of text...
But it does look like some of the above can be simplified. For example `LinkIconCircle` can contain 
```
  --leo-icon-size: 16px;
  color: ${leo.color.white};
```
without having to make a whole new LinkIcon class.

Same for `BackButton` (and if you use Leo's `<Button>`, it will automatically set a good `--leo-icon-size` based on the button type.
And maybe something similar would apply to the component that `InfoIcon` is in?

Reply:
Thinking about this a bit more it could be just wrapped in a styled component with props?

```ts
export const LinkIcon = styled.div<{ size: string, color: string }>`
    --leo-icon-size: ${p => p.size};
    color: ${p => p.color};
```

but if we're doing this, maybe we just want to expose it on our button? I think if this is going to be good for developer experience, I'd be in favor of doing it. In addition, it'll make it much easier to tell what the icon is going to look like:

```tsx
<Icon color={leo.color.white} size={leo.icon.size[16] name="link"/>

// as opposed to

const WhatIDoNowIcon = styled(Icon)`
    --leo-icon-size: ${leo.icon.size[16};
    color: ${leo.color.white};
`

/* a whole bunch of code which isn't related to icons at all */

<WhatIDoNowIcon name="link"/>
```

**controversial opinion:**
I guess what I'm saying is that I don't see the harm in adding these style properties, if they're going to be super common